meilisearch-core/src/store/main.rs

```diff
@@ -28,6 +28,7 @@ const SCHEMA_KEY: &str = "schema";
 const SORTED_DOCUMENT_IDS_CACHE_KEY: &str = "sorted-document-ids-cache";
 const STOP_WORDS_KEY: &str = "stop-words";
 const SYNONYMS_KEY: &str = "synonyms";
+const SYNONYMS_UNICASED_KEY: &str = "synonyms-unicased";
```

Comment:
Why do you keep both?

Reply:
so the synonyms retrieved are the same as the one that were put in

Reply:
I think we should transparently assume that we transform synonyms, no ?

lets say I add,
```
"synonyms": {
            "wolverine": ["xmen", "logan"],
            "wolvérine": ["filin", "toto"]
        }
```

It will be displayed as:

```
"synonyms": {
            "wolverine": ["xmen", "logan"],
            "wolvérine": ["filin", "toto"]
        }
```

but behave as:

```
"synonyms": {
            "wolverine": ["xmen", "logan", "filin", "toto"]
        }
```

am I right ?

Reply:
uhmm, I hadn't thought of that. It actually replace the old one, but we would expect it to behave as:
```
"synonyms": {
            "wolverine": ["xmen", "logan", "filin", "toto"]
        }
```
indeed. This does not answer your question though... What are your opinions @curquiza @qdequele @Kerollmops ?

Reply:
I am more on the many side here, I prefer when behaviors are explicit, and here we will 1. remove some code and 2. make it explicit to the user that we always transform syninyms.

Reply:
and what should be the behaviour of this?
```
"synonyms": {
            "wolverine": ["xmen", "logan"],
            "wolvérine": ["filin", "toto"]
        }
```

Reply:
I think we should just merge them as wolverine is the same as wolvérine for the engine, for now at least, maybe in the future we will change that (e.g. for chinese, thaï).

Reply:
Hello sorry for the delay!
I agree with Kero and Many for the moment, maybe it's better to assume this behaviorm for a first try.